src/sdk/pynni/nni/compression/tensorflow/builtin_pruners.py

```diff
@@ -155,40 +155,35 @@ def calc_mask(self, layer, config):
             return self.mask_dict.get(layer.name)
 
         try:
-            weight = tf.stop_gradient(tf.transpose(weight, [2, 3, 0, 1]))
-            masks = np.ones(weight.shape)
-
-            num_kernels = weight.shape[0] * weight.shape[1]
+            w = tf.stop_gradient(tf.transpose(tf.reshape(weight, (-1, weight.shape[-1])), [1, 0]))
+            masks = np.ones(w.shape)
+            num_kernels = w.shape[0]
```

Comment:
as fpgm is short for Filter Pruning via Geometric Median, maybe rename `num_kernels`->`num_filters` could be more clear

Reply:
ok